tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -1,170 +1,154 @@
+  const renderWithDialog = ({ ...otherProps } = {}) => {
+    render(otherProps);
 
     // Click to open ConfirmationDialog.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
-
-    return root;
-  };
-
-  const getDialogProp = (root, propName) => {
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp(propName);
-    return confirmDialog.prop(propName);
+    userEvent.click(
+      screen.getByRole('button', {
+        name: defaultChildText,
+      }),
+    );
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
   };
 
   it('renders a button', () => {
-    const root = render();
+    render();
 
-    expect(root).toHaveClassName('ConfirmButton');
-    expect(root.find(Button)).toHaveLength(1);
-    expect(root.find(Button)).toHaveClassName('ConfirmButton-default-button');
-    expect(root.find(Button)).toHaveProp('buttonType', 'neutral');
+    const button = screen.getByRole('button', {
+      name: defaultChildText,
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--neutral');
   });
 
   it('passes the buttonType prop to the button', () => {
-    const buttonType = 'alert';
-    const root = render({ buttonType });
+    render({ buttonType: 'alert' });
 
-    expect(root.find(Button)).toHaveProp('buttonType', buttonType);
+    const button = screen.getByRole('button', {
+      name: defaultChildText,
+    });
+    expect(button).toHaveClass('Button--alert');
   });
 
   it('passes the children prop to the button', () => {
     const children = 'Do you really want to delete this?';
-    const root = render({ children });
+    render({ children });
 
-    expect(root.find(Button).children()).toHaveText(children);
+    expect(screen.getByRole('button', { name: children })).toBeInTheDocument();
   });
 
   it('shows ConfirmationDialog when button is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
+    render();
 
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
+    expect(
+      screen.queryByClassName('ConfirmationDialog'),
+    ).not.toBeInTheDocument();
+    expect(screen.queryByText('some warning message')).not.toBeInTheDocument();
 
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
+    userEvent.click(
+      screen.getByRole('button', {
+        name: defaultChildText,
+      }),
+    );
 
-    expect(root).toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
+    expect(screen.getByText('some warning message')).toBeInTheDocument();
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
```

Comment:
As above, we don't need this, and any time you find yourself using `getByClassName` you should pause to consider if it is absolutely necessary and try to find an alternative if possible.